packages/engine/tests/units/experiment.rs

```diff
@@ -27,99 +28,211 @@ pub struct ExpectedOutput {
     pub analysis_outputs: Option<Analysis>,
 }
 
-pub enum Experiment {
-    Simple { experiment: String },
-    SingleRun { steps: u64 },
-}
+fn load_manifest<P: AsRef<Path>>(project_path: P, language: Option<Language>) -> Result<Manifest> {
+    let project_path = project_path.as_ref();
 
-impl fmt::Display for Experiment {
-    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Experiment::Simple { experiment } => write!(fmt, r#"experiment "{experiment}""#),
-            Experiment::SingleRun { steps } => write!(fmt, "experiment with {steps} steps"),
-        }
+    let experiments_json = project_path.join("experiments.json");
+    let src_folder = project_path.join("src");
+    let behaviors_folder = src_folder.join("behaviors");
+    let globals_json = src_folder.join("globals.json");
+    let views_folder = project_path.join("views");
+    let analysis_json = views_folder.join("analysis.json");
+    let data_folder = project_path.join("data");
+
+    let mut project = Manifest::new();
+
+    let (init_js, init_py, init_json) = match language {
+        Some(Language::JavaScript) => (
```

Comment:
I don't think I quite understand what's going on here. So we definitely need to add comments.

Is it that these are inits _for_ javascript tests or something? So `init-js.json` is going to have js behaviors, but we're just initialising it through json?

Reply:
If we're allowing per-language initialisation, and this is very specifically for our tests (and isn't something we're exposing outside of the test suite), then I'm not sure we need the hierarchy of `js` > `py` > `json` right here. I don't see a case where we'd want to supply multiple within the same folder. It might be clearer if it either asserts there's only one and uses that, or if the argument on the macro is the name of the init file rather than a language variant

Reply:
I simplified the logic and let the function fail as soon as multiple initial states were provided.

Reply:
Think that's a lot better